spec/lib/api/v3/work_packages/work_package_representer_spec.rb

```diff
@@ -325,6 +325,26 @@
         end
       end
 
+      describe 'revisions' do
+        context 'when the user lacks the view_changesets permission' do
+          it 'should not have a link to revisions' do
+            expect(subject).not_to have_json_path('_links/revisions/href')
+          end
```

Comment:
You should be using the `has no link` helper here too.

In case anyone decides to embed the repositories, this will automatically make sure that it is not possible to forget the permission check for those (because "has no link" ensures that nothing is embedded under `_embedded/revisions`).